File: win-prog/ProgWinCS/Images and Bitmaps/ImageReflection/ImageReflection.cs
//----------------------------------------------
// ImageReflection.cs © 2001 by Charles Petzold
//----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ImageReflection: PrintableForm
{
     Image image;

     public new static void Main()
     {
          Application.Run(new ImageReflection());
     }
     public ImageReflection()
     {
          Text = "Image Reflection";

          image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          int cxImage = image.Width;
          int cyImage = image.Height;

          grfx.DrawImage(image, cx / 2, cy / 2,  cxImage,  cyImage);
          grfx.DrawImage(image, cx / 2, cy / 2, -cxImage,  cyImage);
          grfx.DrawImage(image, cx / 2, cy / 2,  cxImage, -cyImage);
          grfx.DrawImage(image, cx / 2, cy / 2, -cxImage, -cyImage);
     }
}

File: win-prog/ProgWinVB/Images and Bitmaps/ImageReflection/ImageReflection.vb
'------------------------------------------------
' ImageReflection.vb (c) 2002 by Charles Petzold
'------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ImageReflection
    Inherits PrintableForm

    Private img As Image

    Shared Shadows Sub Main()
        Application.Run(New ImageReflection())
    End Sub

    Sub New()
        Text = "Image Reflection"
        img = Image.FromFile("..\..\Apollo11FullColor.jpg")
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.DrawImage(img, cx \ 2, cy \ 2, img.Width, img.Height)
        grfx.DrawImage(img, cx \ 2, cy \ 2, -img.Width, img.Height)
        grfx.DrawImage(img, cx \ 2, cy \ 2, img.Width, -img.Height)
        grfx.DrawImage(img, cx \ 2, cy \ 2, -img.Width, -img.Height)
    End Sub
End Class
